### ISSUE TITLE
Improve the UX of adding the web-diaspora:// protocol handler

### ISSUE THREAD

**Issue by denschub** [diaspora/diaspora#7832]
PR #7826 implemented a call to `navigator.registerProtocolHandler` to register a handler for `web+diaspora://` URIs. Unfortunately, the implementation is not quite there yet, and probably needs some improvement:

* While we do not encourage it, we do allow non-https installations. In this case, the call to `registerProtocolHandler` would throw a `SecurityError`, possibly breaking the rest of the JS. Even if we ignore non-http pods, there are other cases where the call can throw - for example if the UA blocks the registration of any protocol handlers. TorBrowser, for example, is an example of a UA blocking all protocol handlers. We should at least catch the exception and throw it away.
* Unfortunately, browsers are weird and specs are even weirder. There are no UI implementation rules for custom protocol handlers, so stuff differs. In Chrome, we only get a rather unobtrusive icon in the address bar, while on Firefox, we receive a full-width bar below the address bar, asking the user to enable the handler.
  * In Chrome, I am concerned that people will never see the icon, and thus never enable the handler, which would limit our ability to use it, and that sounds sad.
  * In Firefox, everyone will notice it, for sure. Unfortunately, if the user decides not to install the handler and click the `x` button... They will get asked again on the very next request, which makes this really annoying.
* Although the representation differs across browsers, they have one thing in common: they are very indescriptive, and users have no idea what it means. In fact, some are confused by it and I already got several questions on both public and private channels.

As these issues are kinda large, and the PR got merged into next-minor, I'm marking this as blocking. Sorry for not looking at the PR at an earlier stage.

Based on the current situation, I have two possible solutions in my mind:

1. Do not attempt to register the handler on every page load. Instead, add a button to the user settings page calling the register function, and put some explanation next to it.
2. Do not directly try to register, but show a modal dialog explaining the protocol handler and asking the user if they want it. If they say yes, we can call out to `registerProtocolHandler` and be happy. Otherwise, we should store that decision somewhere and never ask the user again.

Given that idea 2 is a huge effort, I'd not suggest that as a solution for now. As we have no way of generating the links inside the UI anyway, and are not using them in the UI, it feels more like an experimental thing, so having it behind a button in the settings page may be fine for now. That would allow us to ship the handler, but without annoying our users by accident and without having to spend a lot of effort in a more elaborate UI solution.

/cc @SuperTux88 and @cmrd-senya

**Comment by SuperTux88** [comment]
> While we do not encourage it, we do allow non-https installations. In this case, the call to `registerProtocolHandler` would throw a `SecurityError`, possibly breaking the rest of the JS.

Oh, I didn't know that non-https would be a problem, I even tested it on my dev-pod without https and it worked perfectly, but maybe `localhost` is something special?

> for example if the UA blocks the registration of any protocol handlers.

I also didn't know that there are UA which have the function, but block it. I expected that every browser not supporting protocol handlers just doesn't have the function. Why should they add the function and then throw an exception?

> In Chrome, I am concerned that people will never see the icon, and thus never enable the handler, which would limit our ability to use it, and that sounds sad.

I saw that, but thought that maybe chrome changes this in the future or we could add a popup and draw attention to it later (like you described in your second idea).

> In Firefox, everyone will notice it, for sure. Unfortunately, if the user decides not to install the handler and click the x button... They will get asked again on the very next request, which makes this really annoying.

I thought about that, because I thought it maybe would be a problem if you have accounts on multiple pods and I thought you can only install one handler for it and they would constantly switch between the pods because every pod would constantly register it again. But I tested that and found that you can add as many pods you want and select when clicking a link or set one pod as default. So I thought that people could just add it and if they want they could set one as default. I didn't think that people maybe don't want to add it, because they could just add it and when they don't use it, it wouldn't harm, so why not add it? But now I think that that was probably a wrong decision and asking the user only once (and after that only in the settings) would be a much better solution, like you described it.
